ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift

```diff
@@ -206,6 +196,76 @@ struct AIChatResponseMessageView: View {
 
     return Text(text + period + space + learnMoreLink)
   }
+
+  /// View for displaying a Code Block with a header displaying language hint & copy button,
+  /// line numbers and the highlighted code.
+  struct CodeBlockView: View {
+
+    let block: MarkdownParser.StringBlock
+    let codeBlock: MarkdownParser.CodeBlock
+
+    var body: some View {
+      VStack(spacing: 0) {
+        headerSection
+          .background(Color(braveSystemName: .pageBackground))
+        codeBlockSection
+          .background(codeBlock.backgroundColor)
+      }
+      .clipShape(RoundedRectangle(cornerRadius: 8.0, style: .continuous))
+      .overlay(
+        RoundedRectangle(cornerRadius: 8)
+          .inset(by: 0.5)
+          .stroke(Color(braveSystemName: .dividerSubtle), lineWidth: 1)
+      )
+    }
+
+    private var headerSection: some View {
+      HStack {
+        Text(
+          codeBlock.languageHint?.capitalizeFirstLetter ?? Strings.AIChat.leoCodeExampleDefaultTitle
+        )
+        .foregroundStyle(Color(braveSystemName: .textPrimary))
+        Spacer()
+        Button(
+          action: {
+            UIPasteboard.general.setValue(
+              String(block.string.characters),
+              forPasteboardType: "public.plain-text"
+            )
```

Comment:
Can we not use `UIPasteboard.general.string = block.string`?

Reply:
I was matching previous styling in AIChat target, but I will update those too for readability 👍.

Reply:
Updated in [26f2214](https://github.com/brave/brave-core/pull/24606/commits/26f22148aaa1852ca6765c183d745b03dd31e685)